rgw/v2/tests/s3_swift/test_cloud_transition.py

```diff
@@ -0,0 +1,198 @@
+
+<input_yaml>:
+    configs/test_cloud_transitions.yaml
+"""
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import argparse
+import json
+import logging
+import time
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from botocore.exceptions import ClientError
+from v2.lib.exceptions import RGWBaseException
+from v2.lib.resource_op import Config
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.tests.s3_swift.reusables import lc_policy
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import HttpResponseParser, RGWService
+
+log = logging.getLogger()
+TEST_DATA_PATH = None
+
+
+def test_exec(config, ssh_con):
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    ceph_conf = CephConfOp(ssh_con)
+    rgw_service = RGWService()
+
+    user_info = s3lib.create_users(1)[0]
+    # authenticate
+    auth = Auth(user_info, ssh_con, ssl=config.ssl)
+    rgw_conn = auth.do_auth()
+
+    ceph_conf = CephConfOp(ssh_con)
+    log.info("making changes to ceph.conf")
+    ceph_conf.set_to_ceph_conf(
+        "global",
+        ConfigOpts.rgw_lc_debug_interval,
+        str(config.rgw_lc_debug_interval),
+        ssh_con,
+    )
+
+    log.info("Create buckets and objects in source cluster")
+    if config.test_ops["create_bucket"] is True:
+        log.info("no of buckets to create: %s" % config.bucket_count)
+        for bc in range(config.bucket_count):
+            bucket_name_to_create = utils.gen_bucket_name_from_userid(
+                user_info["user_id"], rand_no=bc
+            )
+            log.info("creating bucket with name: %s" % bucket_name_to_create)
+            bucket = reusable.create_bucket(bucket_name_to_create, rgw_conn, user_info)
+            if config.test_ops["create_object"] is True:
+                # uploading data
+                log.info("s3 objects to create: %s" % config.objects_count)
+                config.mapped_sizes = utils.make_mapped_sizes(config)
+                for oc, size in list(config.mapped_sizes.items()):
+                    config.obj_size = size
+                    s3_object_name = utils.gen_s3_object_name(bucket_name_to_create, oc)
+                    log.info("s3 object name: %s" % s3_object_name)
+                    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                    log.info("s3 object path: %s" % s3_object_path)
+                    log.info("upload type: normal")
+                    reusable.upload_object(
+                        s3_object_name, bucket, TEST_DATA_PATH, config, user_info
+                    )
+
```

Comment:
It looks like we are adding support of normal upload.
@vivekanandan-k-rh CAn we include multipart upload as well for more test coverage

Reply:
@udaysk23 , I have a ticket available to cover multipart transition and corresponding validations: https://issues.redhat.com/browse/RHCEPHQE-5925. Will update this code while working on this ticket in coming sprint